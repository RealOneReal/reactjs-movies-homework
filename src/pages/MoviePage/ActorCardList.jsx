import { Grid } from "@mui/material";
import ActorCard from "../../components/ActorCard/ActorCard";
import { credits } from "../../mock-data/mockMovieDetail";

const ActorCardList = () => {
  const cast = credits.cast?.slice(0, 5);

  return (
    <Grid container spacing={3} sx={{ mt: 2 }}>
      {cast &&
        cast.map((card) => (
          <Grid item key={card.id}>
            <ActorCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};

export default ActorCardList;
