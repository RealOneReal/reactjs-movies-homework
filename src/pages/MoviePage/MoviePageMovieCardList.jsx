import { Grid } from "@mui/material";
import MovieCard from "../../components/MovieCard/MovieCard";
import DataFromTMDB from "../../mock-data/mockMovies";

const MoviesPageMovieCardList = () => {
  // todo: get recommendations from Redux ,it`s temporary implementation

  const results = DataFromTMDB?.results.slice(0, 5);
  return (
    <Grid container spacing={1} justifyContent="center">
      {results &&
        results.map((card) => (
          <Grid item key={card.id}>
            <MovieCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};

export default MoviesPageMovieCardList;
