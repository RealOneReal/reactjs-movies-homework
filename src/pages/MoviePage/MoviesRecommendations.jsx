import MovieCardList from "../../components/MovieCardList/MovieCardList";
import DataFromTMDB from "../../mock-data/mockMovies";

const MoviesRecommendations = () => {
  // todo: get recommendations from Redux ,it`s temporary implementation

  const results = DataFromTMDB?.results.slice(0, 5);
  return <MovieCardList {...{ results }} />;
};

export default MoviesRecommendations;
