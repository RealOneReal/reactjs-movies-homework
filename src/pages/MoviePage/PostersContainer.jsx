import { Grid } from "@mui/material";
import { images } from "../../mock-data/mockMovieDetail";

const PostersContainer = () => {
  const posters = images.posters?.slice(0, 8);

  return (
    <Grid container gap={1}>
      {posters.map((img, index) => (
        <Grid
          item
          key={index}
          sx={{
            backgroundImage: `url(${img.filePath})`,
            backgroundPosition: "center center",
            backgroundSize: "contain",
            backgroundRepeat: "no-repeat",
            width: "210px",
            height: "120px",
          }}
        />
      ))}
    </Grid>
  );
};

export default PostersContainer;
