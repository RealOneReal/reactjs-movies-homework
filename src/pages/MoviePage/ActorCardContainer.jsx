import { Box } from "@mui/system";
import ActorCardList from "../../components/ActorCardList/ActorCardList";
import { credits } from "../../mock-data/mockMovieDetail";

const ActorCardContainer = () => {
  const cast = credits.cast?.filter((_item, index) => index < 5);

  return (
    <Box sx={{ mt: 2 }}>
      <ActorCardList {...{ cast }} />
    </Box>
  );
};

export default ActorCardContainer;
