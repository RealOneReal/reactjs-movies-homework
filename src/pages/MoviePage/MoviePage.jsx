import { Box, Container, Typography } from "@mui/material";
import MoviesRecommendations from "./MoviesRecommendations";
import PostersContainer from "./PostersContainer";
import ActorCardContainer from "./ActorCardContainer";

const MoviePage = ({
  posterPath,
  voteAverage,
  title,
  overview,
  releaseDate,
  revenue,
  runtime,
  genres,
}) => {
  const duration = `${Math.floor(runtime / 60)}:${runtime % 60}`;
  const movieGenres = genres?.map((item) => `${item.name} `);

  const styles = {
    main_section: {
      display: "flex",
      gap: 4,
      flexDirection: {
        xs: "column",
        sm: "row",
      },
    },
    poster: {
      backgroundImage: `url(${posterPath})`,
      backgroundPosition: "center",
      backgroundSize: "cover",
      width: "250px",
      height: "375px",
      objectFit: "contain",
      borderRadius: "10px",
      position: "relative",
      margin: "auto",
    },
    poster_vote: {
      backgroundColor: "green",
      color: "primary.light",
      width: "50px",
      height: "30px",
      textAlign: "center",
      position: "absolute",
      left: "-10px",
      top: "20px",
      borderRadius: "5px",
    },
  };

  return (
    <Box sx={{ backgroundColor: "secondary.dark" }}>
      <Container maxWidth="lg" sx={{ pt: 2 }}>
        <Box className="main-section" sx={styles.main_section}>
          <Box className="main-section_movie">
            <Box sx={styles.poster}>
              <Typography component="div" variant="h6" sx={styles.poster_vote}>
                {voteAverage}
              </Typography>
            </Box>
          </Box>
          <Box className="main-section_movie-items">
            <Typography variant="subtitle2">Title:</Typography>
            <Typography variant="h3">{title}</Typography>
            <Typography variant="subtitle2">Overview:</Typography>
            <Typography variant="body1">{overview}</Typography>
            <Typography variant="subtitle2">Release date:</Typography>
            <Typography variant="body1">{releaseDate}</Typography>
            <Typography variant="subtitle2">Revenue:</Typography>
            <Typography variant="body1">{`$ ${revenue}`}</Typography>
            <Typography variant="subtitle2">Duration:</Typography>
            <Typography variant="body1">{duration}</Typography>
            <Typography variant="subtitle2">Genres:</Typography>
            <Typography variant="body1">{movieGenres}</Typography>
            <Typography variant="h6">Top Billed Cast:</Typography>
            <ActorCardContainer />
            <Typography variant="h6">Images:</Typography>
            <PostersContainer />
          </Box>
        </Box>
        <Box className="recommendations">
          <Typography variant="h5" color="primary.dark">
            RECOMMENDATIONS :
          </Typography>
          <MoviesRecommendations></MoviesRecommendations>
        </Box>
      </Container>
    </Box>
  );
};

export default MoviePage;
