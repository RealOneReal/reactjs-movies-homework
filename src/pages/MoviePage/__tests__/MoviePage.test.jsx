import { render } from "@testing-library/react";
import MoviePage from "../MoviePage";
import MoviesRecommendations from "../MoviesRecommendations";
import PostersContainer from "../PostersContainer";
import ActorCardContainer from "../ActorCardContainer";

jest.mock("../MoviesRecommendations", () => () => "MoviesRecommendations");
jest.mock("../PostersContainer", () => () => "PostersContainer");
jest.mock("../ActorCardContainer", () => () => "ActorCardContainer");

const mockData = {
  posterPath: "/123",
  voteAverage: 5,
  title: "Matrix",
  overview: "nice movie",
  releaseDate: "18.02.1998",
  revenue: "1.000.000",
  runtime: 300,
  genres: [{ name: "comedy" }, { name: "action" }],
};

describe("MoviePage", () => {
  it("should be rendered", () => {
    const { container } = render(<MoviePage {...mockData} />);
    expect(container).toMatchSnapshot();
  });
});
