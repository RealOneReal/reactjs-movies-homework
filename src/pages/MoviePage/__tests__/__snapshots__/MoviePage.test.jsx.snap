// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should be rendered 1`] = `
<div>
  <div
    class="MuiBox-root css-4ozt90"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-r5r38p-MuiContainer-root"
    >
      <div
        class="main-section MuiBox-root css-n0klh3"
      >
        <div
          class="main-section_movie MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-gittgl"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-8c5bzc-MuiTypography-root"
            >
              5
            </div>
          </div>
        </div>
        <div
          class="main-section_movie-items MuiBox-root css-0"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Title:
          </h6>
          <h3
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Matrix
          </h3>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Overview:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            nice movie
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Release date:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            18.02.1998
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Revenue:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            $ 1.000.000
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Duration:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            5:0
          </p>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
          >
            Genres:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            comedy 
            action 
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Top Billed Cast:
          </h6>
          ActorCardList
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Images:
          </h6>
          PostersContainer
        </div>
      </div>
      <div
        class="recommendations MuiBox-root css-0"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-1jo76pw-MuiTypography-root"
        >
          RECOMMENDATIONS :
        </h5>
        MoviesPageMovieCardList
      </div>
    </div>
  </div>
</div>
`;
